DigitalLearningSolutions.Data.Tests/DataServices/CompetencyLearningResourcesDataServiceTests.cs

```diff
@@ -0,0 +1,85 @@
+﻿namespace DigitalLearningSolutions.Data.Tests.DataServices
+{
+    using System.Transactions;
+    using DigitalLearningSolutions.Data.DataServices;
+    using DigitalLearningSolutions.Data.Models.LearningResources;
+    using DigitalLearningSolutions.Data.Tests.TestHelpers;
+    using FluentAssertions;
+    using NUnit.Framework;
+
+    public class CompetencyLearningResourcesDataServiceTests
+    {
+        private ICompetencyLearningResourcesDataService service = null!;
+        private CompetencyLearningResourcesTestHelper testHelper = null!;
+
+        [SetUp]
+        public void Setup()
+        {
+            var connection = ServiceTestHelper.GetDatabaseConnection();
+            service = new CompetencyLearningResourcesDataService(connection);
+
+            testHelper = new CompetencyLearningResourcesTestHelper(connection);
+        }
+
+        [Test]
+        public void GetCompetencyLearningResourceById_returns_expected_resource()
+        {
+            using var transaction = new TransactionScope();
+            try
+            {
+                // Given
+                var expectedResource = new CompetencyLearningResource
+                {
+                    Id = 1,
+                    CompetencyId = 1,
+                    LearningHubResourceReference = 2,
+                    AdminId = 7,
+                };
+                InsertCompetencyLearningResources();
+
+                // When
+                var result = service.GetCompetencyLearningResourceById(1);
+
+                // Then
+                result.Should().BeEquivalentTo(expectedResource);
+            }
+            finally
+            {
+                transaction.Dispose();
+            }
+        }
+
+        [Test]
+        public void GetCompetencyIdsByLearningHubResourceReference_returns_expected_ids()
+        {
+            using var transaction = new TransactionScope();
+            try
+            {
+                // Given
+                var expectedIds = new[] { 1, 2, 3, 5, 6 };
+                InsertCompetencyLearningResources();
+
+                // When
+                var result = service.GetCompetencyIdsByLearningHubResourceReference(2);
+
+                // Then
+                result.Should().BeEquivalentTo(expectedIds);
+            }
+            finally
+            {
+                transaction.Dispose();
+            }
+        }
+
+        private void InsertCompetencyLearningResources()

```

Comment:
I don't see any sensible way to do it without identity insert since we won't be able to get the item by Id since the Id will auto-increment.

Given it is against the test database though, I don't think it'll matter as long as we assume we always insert records in transactions like we are doing here (i.e. no permanent test data ever) or we create some permanent test data right now.

Otherwise I think the only options that would be more future proof would be to insert them at the max int value (unlikely to hit this) or delete all data in the transaction (which would likely violate some foreign keys if we did have data)